starlark/eval_test.go

```diff
@@ -966,6 +972,62 @@ func TestCancel(t *testing.T) {
 		if fmt.Sprint(err) != `Starlark computation cancelled: "nope"` {
 			t.Errorf("execution returned error %q, want cancellation", err)
 		}
+		if !errors.Is(err, starlark.ErrExecutionCancelled) {
+			t.Errorf("execution didn't return error starlark.ErrExecutionCancelled")
+		}
+	}
+}
+
+func TestCancelWithError(t *testing.T) {
+	// A thread cancelled before it begins executes no code.
+	{
+		thread := new(starlark.Thread)
+		cancelErr := errors.New("nope")
+		thread.CancelWithError(cancelErr)
+		_, err := starlark.ExecFile(thread, "precancel.star", `x = 1//0`, nil)
+		if fmt.Sprint(err) != "Starlark computation cancelled: nope" {
+			t.Errorf("execution returned error %q, want cancellation", err)
+		}
+		if !errors.Is(err, starlark.ErrExecutionCancelled) {
+			t.Errorf("execution didn't return error starlark.ErrExecutionCancelled")
+		}
+		if !errors.Is(err, cancelErr) {
+			t.Errorf("execution didn't return expected error")
+		}
+
+		// cancellation is sticky
+		_, err = starlark.ExecFile(thread, "precancel.star", `x = 1//0`, nil)
+		if fmt.Sprint(err) != "Starlark computation cancelled: nope" {
+			t.Errorf("execution returned error %q, want cancellation", err)
+		}
+		if !errors.Is(err, starlark.ErrExecutionCancelled) {
+			t.Errorf("execution didn't return error starlark.ErrExecutionCancelled")
```

Comment:
Let's make the assertions more specific about what exactly they tested and what they got, e.g. `"ExecFIle: !error.Is(ErrCanceled): got %T, %v", err, err`

Reply:
Will do so.

Reply:
Done.